src/integrations/hapiApollo.ts

```diff
@@ -137,9 +152,75 @@ async function processQuery(body, optionsObject) {
       responses.push({ errors: [formatErrorFn(e)] });
     }
   }
-  return responses;
+  return reply(responses);
 }
 
 function isOptionsFunction(arg: ApolloOptions | HAPIOptionsFunction): arg is HAPIOptionsFunction {
   return typeof arg === 'function';
 }
+
+function createErr(code: number, message: string) {
+  const err = Boom.create(code);
+  err.output.payload.message = message;
```

Comment:
I just double check and you are right. For an error 500, the message is overwritten.
See [https://github.com/hapijs/boom/blob/master/lib/index.js#L136-L138](https://github.com/hapijs/boom/blob/master/lib/index.js#L136-L138)